File: typescript/ch1/power.ts
const array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];

function power_ip() {
  const result: number[] = [];
  for (let i = 0; i < array.length; i++) {
    result.push(Math.pow(array[i], 2));
  }

  console.log('ipPower', result);
}

const power_fp = () => console.log('fpPower', array.map((num) => Math.pow(num, 2)));

export {
  power_ip,
  power_fp,
};

File: python/python/ch1/power.py
import math
import pydash

array = range(10)

def power_ip():
    result = []
    for data in array:
        result.append(math.pow(data, 2))
        
    print('power_ip', result)
    
power_fp = lambda: print('power_fp', pydash.map_(array, lambda num: math.pow(num, 2)))

power = [
    power_ip,
    power_fp
]